packages/auth-sdk/user.tsx

```diff
@@ -16,16 +17,46 @@ const userKeys = {
    user: ['user'],
 };
 
+const hasLocalStorage = typeof localStorage !== 'undefined';
+const userDataLocalKey = 'user.data';
+
 export function useAuthenticatedUser() {
    const apiClient = useIFixitApiClient();
+   const cachedUserData = useMemo(
+      () =>
+         !hasLocalStorage
+            ? null
+            : (() => {
+                 const storedString = localStorage.getItem(userDataLocalKey);
+                 return storedString === null ? null : JSON.parse(storedString);
+              })(),
+      []
+   );
    const query = useQuery(
       userKeys.user,
-      () => fetchAuthenticatedUser(apiClient).catch(() => null),
+      () => {
+         const responsePromise = fetchAuthenticatedUser(apiClient).catch(
+            () => null
+         );
+         if (hasLocalStorage) {
+            responsePromise
+               .then((response) => {
+                  localStorage.setItem(
+                     userDataLocalKey,
+                     JSON.stringify(response)
+                  );
+               })
+               .catch(() => {});
```

Comment:
I'm curious why this extra catch block is here

Reply:
Calling methods on `localStorage` sometimes throws exceptions if they have some browser security feature enabled:
https://sentry.io/organizations/ifixit/issues/3505592338/?query=is%3Aunresolved+localstorage&referrer=issue-stream&statsPeriod=14d

And I don't want those reported to sentry.

Reply:
Makes sense!